src/main/include/subsystems/BaseSingleAxisSubsystem.h

```diff
@@ -0,0 +1,372 @@
+      return;
+    } else {
+      if (_log)
+        Logging::logToStdOut(_prefix,
+                             "SETTING SPEED TO: " + std::to_string(speed),
+                             Logging::Level::VERBOSE, _ansiPrefixModifiers);
+      _motor.Set(speed);
+    }
+  }
+
+  void RunMotorSpeedDefault(bool invertDirection = false) override {
+    RunMotorSpeed(invertDirection ? -_config.defaultMovementSpeed
+                                  : _config.defaultMovementSpeed);
+  }
+
+  /**
+   * @brief Call this one from a joystick-bound command to override current
+   * movement
+   *
+   * @param speed Percentage speed
+   */
+  void RunMotorExternal(double speed) override {
+    // TODO: constant
+    if (abs(speed) <= 0.05) {
+      if (_isMovingToPosition)
+        return; // Don't set the motor and overwrite a potential
+                // automated movement
+
+      // TODO: Use velocity PID instead
+      if (_config.type == AxisType::Rotational)
+        RunMotorSpeed(ArmConstants::kAntiGravityPercentage); // Make 'er hover!
+      else
+        _motor.Set(0);
+
+      return;
+    }
+
+    // Overwrite current automated position with joystick input
+    if (_isMovingToPosition) {
+      StopMovement();
+    }
+
+    RunMotorSpeed(speed);
+  }
+
+  void JoystickMoveStep(double rotation) override {
+    auto steps = rotation * _config.stepSize;
+    auto newTarget = IncrementTargetPosition(steps);
+    MoveToPosition(newTarget);
+  }
+
+  double IncrementTargetPosition(double steps) override {
+    return std::clamp(_targetPosition + steps, _config.minDistance,
+                      _config.maxDistance);
+  }
+
+  void UpdateMovement() override {
+    if (_isMovingToPosition) {
+      if (_log)
+        Logging::logToStdOut(
+            _prefix,
+            "Target Position: " + std::to_string(_targetPosition) +
+                std::string(_config.type == AxisType::Linear ? " in" : " deg"),
+            Logging::Level::INFO, _ansiPrefixModifiers);
+
+      auto res = _controller.Calculate(GetCurrentPosition(), _targetPosition) *
+                 _config.pidResultMultiplier;
+      auto clampedRes = std::clamp(res, -1.0, 1.0);
+      if (_log)
+        Logging::logToStdOut(_prefix,
+                             "Clamped Res: " + std::to_string(clampedRes),
+                             Logging::Level::INFO, _ansiPrefixModifiers);
+      Logging::logToSmartDashboard(_prefix + " TargetPos",
+                                   std::to_string(_targetPosition),
+                                   Logging::Level::INFO);
+
+      if (_controller.AtSetpoint()) {
+        Logging::logToStdOut(_prefix, "REACHED GOAL", Logging::Level::INFO,
+                             _ansiPrefixModifiers);
+        StopMovement();
+        return;
+      }
+
+      RunMotorSpeed(clampedRes);
+    }
+  }
+
+  virtual void ResetEncoder() = 0;
+
+  virtual double GetCurrentPosition() = 0;
+
+  bool AtHome() override {
+    if (_minLimitSwitch) {
```

Comment:
Why is limit switch used in AtHome? Do we not have encoder working yet? A comment would be good to explain why AtHome is using AtLimitSwitchHome. I noticed code above was using bool homeState = ignoreEncoder ? AtLimitSwitchHome() : AtHome();

Should we have just one AtHome method that can determine the proper check to make?

Reply:
We always want the limit switch to override anything else first, as it's the last line of defense, which is why we check it first if configured/enabled. Theoretically, this should be the only method we need to call, but sometimes the encoder doesn't work properly and needs to be ignored, hence the ternary expression you mentioned.